onnx/test/checker_test.py

```diff
@@ -89,6 +89,24 @@ def test_check_graph(self):  # type: () -> None
         graph.initializer[0].name = 'X'
         checker.check_graph(graph)
 
+    def test_check_graph_duplicate_init_names(self):  # type: () -> None
```

Comment:
can you add one more tests to test the enforce_has_field behavior for attribute and initializer. This will make the intent explicit.